raft/raft.go

```diff
@@ -1298,6 +1318,11 @@ func stepCandidate(r *raft, m pb.Message) error {
 	case myVoteRespType:
 		gr, rj, res := r.poll(m.From, m.Type, !m.Reject)
 		r.logger.Infof("%x has received %d %s votes and %d vote rejections", r.id, gr, m.Type, rj)
+		if m.Reject && m.Commit > r.raftLog.committed && r.raftLog.matchTerm(m.Commit, m.CommitTerm) {
+			r.logger.Infof("%x [commit: %d, lastindex: %d, lastterm: %d] fast-forwarded commit to vote response [index: %d, term: %d]",
+				r.id, r.raftLog.committed, r.raftLog.lastIndex(), r.raftLog.lastTerm(), m.Commit, m.CommitTerm)
+			r.raftLog.commitTo(m.Commit)
```

Comment:
This breaks an assumption that progress tracker will not change during election. Now we use map to count votes and quorum, so it doesn't seem to affect the library itself.

Reply:
Committing across membership changes during an election is scary. Perhaps whenever the commit index advances this way we revert to being a follower.